File: 1295 - Find Numbers with Even Number of Digits/FindNumbers.js
// https://leetcode.com/problems/find-numbers-with-even-number-of-digits/description/

function findNumbers(nums) {
  let evenNumbers = 0;

  for (let n of nums) {
    let digitCount = 0;

    while (n !== 0) {
      n = Math.floor(n / 10);
      digitCount++;
    }

    if (digitCount % 2 === 0) {
      evenNumbers++;
    }
  }

  return evenNumbers;
}

console.log(findNumbers([14, 3, 26, 12]));

File: 1295 - Find Numbers with Even Number of Digits/FindNumbers.java
// https://leetcode.com/problems/find-numbers-with-even-number-of-digits/description/

public class FindNumbers {
    public static void main(String[] args) {
        System.out.println(findNumbers(new int[] { 14, 3, 26, 12 }));
    }

    public static int findNumbers(int[] nums) {
        int evenNumbers = 0;

        for (int n : nums) {
            int digitCount = 0;

            while (n != 0) {
                n /= 10;
                digitCount++;
            }

            if (digitCount % 2 == 0) {
                evenNumbers++;
            }
        }

        return evenNumbers;
    }
}